### PR TITLE
[RTM] Extract bundle code

### PR DESCRIPTION
**Pull request by aschempp**
There's various stuff currently in `core-bundle` that is actually code from this bundle.

### CODE REVIEW

File: src/EventListener/FileMetaInformationListener.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+     *
+     * @param string $table
+     * @param int    $id
+     *
+     * @return Result|false
+     */
+    public function onAddFileMetaInformationToRequest($table, $id)
+    {
+        if ('tl_news' === $table) {
+            $this->getResult(
```
Comment (ausi) [inline src/EventListener/FileMetaInformationListener.php]: Is there a `return` statement missing here and in line 59?

Comment (aschempp) [inline src/EventListener/FileMetaInformationListener.php]: Ha! Good catch, fixed.

File: src/EventListener/FileMetaInformationListener.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+    /**
+     * Fetches result from database.
+     *
+     * @param string $query
+     * @param mixed  $params
+     *
+     * @return Result|object
+     */
+    private function getResult($query, $params)
```
Comment (Toflar) [inline src/EventListener/FileMetaInformationListener.php:76]: Why don't you inject `connection` and use the new way already here?

Comment (aschempp) [inline src/EventListener/FileMetaInformationListener.php:76]: see https://github.com/contao/core-bundle/pull/471#issuecomment-207379804

File: src/EventListener/FileMetaInformationListener.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+     * Returns the page record related to the given table and ID.
+     *
+     * @param string $table
+     * @param int    $id
+     *
+     * @return Result|false
+     */
+    public function onAddFileMetaInformationToRequest($table, $id)
+    {
+        if ('tl_news' === $table) {
```
Comment (leofeyer) [inline src/EventListener/FileMetaInformationListener.php:49]: Should this be `tl_news_archive`?

**Comment by leofeyer** [comment]
I think we should add unit tests for the listeners, shouldn't we?

File: src/EventListener/FileMetaInformationListener.php (hunk 3 above)
Comment (aschempp) [inline src/EventListener/FileMetaInformationListener.php:49]: It's actually the second call that should be on `tl_news_archive`. Fixed in e596ce7b4bbfa9089993512435a419438a4d272e

**Comment by aschempp** [comment]
Dunno about unit tests, because you would either need a mock database or mock the database class, which means the whole test is rather useless…

**Comment by leofeyer** [comment]
That is probably true, but it would make sense for the insert tags listener and the generate page listener.

**Comment by aschempp** [comment]
That is probably true 😎

**Comment by leofeyer** [comment]
Merged with unit tests added in 51d83263520acdb3f5bbd2e7c56985a1cdb75673.

**Pull request merged**